tests/ActivityTest.php

```diff
@@ -44,24 +44,21 @@ public function testInstantiation() {
 
     public function testFromJSONInvalidNull() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_NONE
+            'TinCan\JSONParseErrorException'
         );
         $obj = Activity::fromJSON(null);
     }
 
     public function testFromJSONInvalidEmptyString() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_NONE
+            'TinCan\JSONParseErrorException'
         );
         $obj = Activity::fromJSON('');
     }
 
     public function testFromJSONInvalidMalformed() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_SYNTAX
+            'TinCan\JSONParseErrorException'
```

Comment:
While a better overall implementation this is a loss of granularity of the unit tests, we no longer test distinguishing malformed versus null/empty. We should maintain that level of detail.